bin/docserv

```diff
@@ -460,16 +447,60 @@ class BuildInstructionHandler:
 
     def __del__(self):
```

Comment:
Be careful, a destructor in Python is usually either a) not needed, and/or b) is not guaranteed it will be executed when you expect it. [StackOverflow](https://stackoverflow.com/a/1481512) says this:

> `__del__` is a destructor.  It is called when an object is garbage collected which happens after all references to the object have been deleted. [...] Since you have no guarantee it's executed, one should never put the code that you need to be run into __del__() — instead, this code belongs to finally clause of the try block or to a context manager in a with statement.

Reply:
Thanks. I added a cleanup() function.